lib/components/field-key/revoke.vue

```diff
@@ -0,0 +1,80 @@
+<!--
+Copyright 2017 Super Adventure Developers
+See the NOTICE file at the top-level directory of this distribution and at
+https://github.com/nafundi/super-adventure/blob/master/NOTICE.
+
+This file is part of Super Adventure. It is subject to the license terms in
+the LICENSE file found in the top-level directory of this distribution and at
+https://www.apache.org/licenses/LICENSE-2.0. No part of Super Adventure,
+including this file, may be copied, modified, propagated, or distributed
+except according to the terms contained in the LICENSE file.
+-->
+<template>
+  <modal :state="state" @hide="$emit('hide')" backdrop
+    :hideable="!awaitingResponse" id="field-key-revoke">
+    <template slot="title">Revoke Key</template>
+    <template slot="body">
+      <alert v-bind="alert" @close="alert.state = false"/>
+      <div class="modal-introduction">
+        <p>
+          Are you sure you want to revoke the field key
+          <strong>{{ fieldKey.displayName }}</strong>?
+        </p>
+        <p>
+          Existing submissions using this key will remain, but anybody relying
+          on this key will have to obtain a new one to continue downloading
+          forms or uploading submissions.
+        </p>
+        <p>This action cannot be undone.</p>
+      </div>
+      <div class="modal-actions">
+        <button type="button" class="btn btn-danger"
+          :disabled="awaitingResponse" @click="revoke">
+          Yes, proceed <spinner :state="awaitingResponse"/>
+        </button>
+        <button type="button" class="btn btn-link" :disabled="awaitingResponse"
+          @click="$emit('hide')">
+          No, cancel
+        </button>
+      </div>
+    </template>
+  </modal>
+</template>
+
+<script>
+import alert from '../../mixins/alert';
+import request from '../../mixins/request';
+
+export default {
+  name: 'FieldKeyRevoke',
+  mixins: [alert(), request()],
+  props: {
+    state: {
+      type: Boolean,
+      default: false
+    },
+    fieldKey: {
+      type: Object,
+      required: true
+    }
+  },
+  data() {
+    return {
+      alert: alert.blank(),
+      requestId: null
+    };
+  },
+  methods: {
+    revoke() {
+      this
+        .delete(`/field-keys/${this.fieldKey.id}`)
```

Comment:
whoa, no. you want to issue a `DELETE` with the `token` to `/v1/sessions`; see https://jubilantgarbanzo.docs.apiary.io/#reference/authentication/field-key-authentication/revoking-a-field-key

Reply:
this must be why you were seeing them disappear